doc/build/sysbuild/index.rst

```diff
@@ -0,0 +1,370 @@
+``--domain`` argument must be used, for example::
+
+  west debug --domain mcuboot
+
+Details on ``west debug`` with sysbuild can be found in
+:ref:`west debug - multi domain flashing <west-multi-domain-debugging>`.
+
+Building a sample with MCUboot
+******************************
+
+Zephyr is compatible with MCUboot, a bootloader that is natively
+supported by sysbuild.
+
+To build a sample, such as ``hello_world``, with MCUboot it is enough to just
+enable MCUboot and build and flash the sample, like this:
+
+.. zephyr-app-commands::
+   :tool: west
+   :app: samples/hello_world
+   :board: reel_board
+   :goals: flash
+   :west-args: --sysbuild
+   :gen-args: -DSB_CONFIG_BOOTLOADER_MCUBOOT=y
+   :compact:
+
+this will build ``hello_world`` and ``mcuboot`` for the ``reel_board``, and then
+flash both ``mcuboot`` and ``hello_world``.
+
+More detailed information regarding the use of MCUboot with Zephyr  can be found
+in the `MCUboot with Zephyr`_ documentation page on the MCUboot website.
+
+.. note::
+   MCUBoot default configuration will perform a full chip erase when flashed.
+   This can be controlled through the MCUBoot ``ZEPHYR_TRY_MASS_ERASE`` Kconfig
+   setting. If this setting is on, then flashing only MCUBoot, for example using
+   ``west flash --domain mcuboot``, may erase the complete flash.
+
+Sysbuild configuration file
+***************************
+
+Sysbuild supports its own configuration file infrastructure similar to Zephyr
+build system.
+
+Sysbuild default looks for a configuration file named ``sysbuild.conf`` in the
+application top-level directory, like this:
+
+.. code-block:: none
+
+   <home>/application
+   ├── CMakeLists.txt
+   ├── prj.conf
+   └── sysbuild.conf
+
+A ``sysbuild.conf`` with the following assignment will enable ``mcuboot`` for
+the application, whenever sysbuild is used for building.
+
+.. code-block:: none
+
+   SB_CONFIG_BOOTLOADER_MCUBOOT=y
+
+Sysbuild also supports custom configuration file names with the
+``-DSB_CONF_FILE=<sysbuild-conf-file>`` CMake build setting.
+
+For example this allows you to create ``sysbuild-mcuboot.conf`` and then specify
+this file instead when building with sysbuild, as:
+
+.. zephyr-app-commands::
+   :tool: west
+   :app: samples/hello_world
+   :board: reel_board
+   :goals: flash
+   :west-args: --sysbuild
+   :gen-args: -DSB_CONF_FILE=sysbuild-mcuboot.conf
+   :compact:
+
+.. _sysbuild_zephyr_application:
+
+Adding Zephyr based application
+*******************************
+
+A board or an application might support additional Zephyr based applications.
+
+Supporting an extra Zephyr based application can be done with a single CMake
+function call in a :file:`CMakeLists.txt` or :file:`*.cmake` file.
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Add(
+     APPLICATION my_sample
+     SOURCE_DIR <path-to>/my_sample
+     BUILD_ALWAYS True
+   )
+
+this will include ``my_sample`` as an extra image in sysbuild, and using the
```

Comment:
-and

Reply:
Done, addressed by https://github.com/zephyrproject-rtos/zephyr/pull/43846#pullrequestreview-1060453559